src/vr.zig

```diff
@@ -2458,16 +2479,16 @@ pub const Replica = struct {
     }
 
     /// The caller owns the returned message, if any, which has exactly 1 reference.
-    fn create_do_view_change_or_start_view_message(self: *Replica, command: Command) *Message {
+    fn create_do_view_change_or_start_view_message(self: *Replica, command: Command) !*Message {
```

Comment:
To reduce dimensionality (because we already have so much with all the interactions between VR `Status` multiplied by VR `view`) I think it's good to avoid having any Replica methods return errors, rather going with `void` or a `pointer` or `null`. If we return an error then we need to switch on it exhaustively at the call site, whereas a null can be `orelse return`. We're making it clear in the API that we'll return a message or nothing vs a message or a set of errors.

Reply:
For this reason, I think we should do the same in `MessagePool`. Instead of returning an error if we can't get a message, just return null. Then the caller knows they don't need to switch further. This is different to what I did in io_uring with `get_sqe()` but I was afraid of nulls then!

Reply:
It's also slightly different to io_uring, because for message passing and within VR we expect to drop messages or retry all the time in line with our fault models, whereas io_uring can't really tell much about the user code running it.